presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -248,7 +260,53 @@ public static boolean canCoerce(Type actualType, Type expectedType)
     public static boolean canCoerce(TypeSignature actualType, TypeSignature expectedType)
     {
         Optional<TypeSignature> commonSuperTypeSignature = getCommonSuperTypeSignature(actualType, expectedType);
-        return commonSuperTypeSignature.isPresent() && commonSuperTypeSignature.get().equals(expectedType);
+        return commonSuperTypeSignature.isPresent() && typesMatch(commonSuperTypeSignature.get(), expectedType);
+    }
+
+    private static boolean typesMatch(TypeSignature actualType, TypeSignature expectedType)
+    {
+        if (!actualType.getBase().equals(expectedType.getBase())) {
+            return false;
+        }
```

Comment:
rename `getCommonSuperTypeForLiteraTypes` to `getCommonSuperTypeForTypesWithLongParameters`